src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -755,7 +756,7 @@ private void AddAllocatedEndpointInfo(IEnumerable<AppResource> resources)
 
                 sp.EndpointAnnotation.AllocatedEndpoint = new AllocatedEndpoint(
                     sp.EndpointAnnotation,
-                    "localhost",
+                    sp.EndpointAnnotation.TargetHost.Replace("0.0.0.0", Dns.GetHostName()).Replace("+", Dns.GetHostName()).Replace("*", Dns.GetHostName()),
```

Comment:
What about `[::]` (the IP v6 equivalent of `0.0.0.0`)?

Reply:
Good call

Reply:
I'll do some additional IPv6 testing on Monday.

Reply:
Also keep in mind that, depending on the API, some like the IPv6 loopback address to be surrounded by brackets `[::]` and some take just `::` 🤷‍♂️